File: app/src/main/java/com/example/agiles_3/ReceipeRepository.java
package com.example.agiles_3;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

class ReceipeRepository {
    private ReceipeDao receipeDao;
    private LiveData<List<Receipe>> allReceipes;

    ReceipeRepository(Application application){
        ReceipeRoomDatabase db = ReceipeRoomDatabase.getDatabase(application);
        receipeDao = db.receipeDao();
        allReceipes = receipeDao.getAllReceipes();
    }

    LiveData<List<Receipe>> getAllReceipes (){
        return allReceipes;
    }

    void insert(Receipe receipe){
        ReceipeRoomDatabase.databaseWriteExecutor.execute(() -> {
            receipeDao.insert(receipe);
        });
    }
}

File: app/src/main/java/com/example/agiles_3/ReceipeRepository.kt
package com.example.agiles_3

import androidx.annotation.WorkerThread
import kotlinx.coroutines.flow.Flow

//Declaramos el Dao como propiedad en el constructor ya que no necesitamos toda la base de datos, solo el DAO
class ReceipeRepository(private val receipeDao: ReceipeDao) {
    //El room hace las peticiones a la db en hilos separados
    //Se ocupa un observador que necesita avisar que se cambiaron los datos

    val allRecipes: Flow<List<Receipe>> = receipeDao.getSortedId()

    //Por default el room suspende las peticiiones del hilo principal a menos que no se necesite implementar nada, para asegurar que no se esta
    //haciendo nada en segundo plano

    @Suppress("RedundantSuspendModifier")
    @WorkerThread
    //Modificar la de insertar
    suspend fun insert(receipe : Receipe){
        receipeDao.insert(receipe)
    }
}